ml-agents/mlagents/trainers/policy/policy.py

```diff
@@ -29,25 +29,27 @@ def __init__(
         condition_sigma_on_obs: bool = True,
     ):
         self.behavior_spec = behavior_spec
+        self.action_spec = behavior_spec.action_spec
         self.trainer_settings = trainer_settings
         self.network_settings: NetworkSettings = trainer_settings.network_settings
         self.seed = seed
         # For mixed action spaces
-        self.continuous_act_size = behavior_spec.continuous_action_size
-        self.discrete_act_size = behavior_spec.discrete_action_branches
+        self.continuous_act_size = self.action_spec.continuous_action_size
```

Comment:
I wonder if we shouldn't store these at all and just call `policy.action_spec.X` in the Optimizer and elsewhere. We can cut down the number of params the Policy needs